### PR TITLE
Code review

### PR DESCRIPTION
**Pull request by jamesjpedi**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,17 +1,14 @@
...
+import React from "react";
+import ReactDOM from "react-dom";
+import "./index.css";
+import App from "./App";
+import { BrowserRouter } from "react-router-dom";
 
 ReactDOM.render(
   <React.StrictMode>
-    <App />
+    <BrowserRouter>
```
Comment (jamesjpedi) [inline src/index.js:9]: Move and manage router in separate file / component

File: src/App.js [hunk 2]
```diff
@@ -1,23 +1,225 @@
-import logo from './logo.svg';
-import './App.css';
+import React from "react";
+import styledComponents from "styled-components";
+import "./App.css";
+import SearchIcons from "./components/SearchIcons/SearchIcons";
+import { Routes, Route, Navigate } from "react-router-dom";
+import Flight from "./pages/Flight";
+import Activities from "./pages/Activities";
+import Hostel from "./pages/Hostel";
```
Comment (jamesjpedi) [inline src/App.js:8]: Refactor import order with react standards

File: src/App.js [hunk 3]
```diff
@@ -1,23 +1,225 @@
...
 
+const Icons = [
```
Comment (jamesjpedi) [inline src/App.js:10]: Move constants to separate file

File: src/App.js [hunk 4]
```diff
@@ -1,23 +1,225 @@
...
+  {
+    id: 10,
+    name: "Activities",
+    icon: "-207px -72px",
+    activeIcon: "-164px -72px",
+    path: "/activities"
+  },
+];
+
+const NavIcon = styledComponents.nav`
```
Comment (jamesjpedi) [inline src/App.js:83]: Move styles to style file

File: src/App.js [hunk 5]
```diff
@@ -1,23 +1,225 @@
...
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+      <Routes>
+        <Route
```
Comment (jamesjpedi) [inline src/App.js:95]: Use standalone component for routes

File: src/App.js [hunk 6]
```diff
@@ -1,23 +1,225 @@
...
+          element={
+            <>
+              <NavIcon>
+                <SearchIcons data={Icons}/>
+              </NavIcon>
+              <Flight labelFor="flight" limit={3}/>
+            </>
+          }
+        />
+        <Route
```
Comment (jamesjpedi) [inline src/App.js:106]: Global routes should move as the last in stack

File: src/pages/Hostel.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
+import React from "react";
+import styledComponents from "styled-components";
+import SearchBox from "../components/SearchBox/SearchBox";
+import SearchButton from "../components/SearchButton/SearchButton";
+import SearchFare from "../components/SearchFare/SearchFare";
+import SearchTrip from "../components/SearchTrip/SearchTrip";
+
+const MainDiv = styledComponents.div`
```
Comment (jamesjpedi) [inline src/pages/Hostel.js:8]: Move styles to style  file

File: src/pages/Flight.js [hunk 8]
```diff
@@ -0,0 +1,32 @@
+import React from "react";
+import styledComponents from "styled-components";
+import SearchBox from "../components/SearchBox/SearchBox";
+import SearchButton from "../components/SearchButton/SearchButton";
+import SearchFare from "../components/SearchFare/SearchFare";
+import SearchTrip from "../components/SearchTrip/SearchTrip";
+
+const MainDiv = styledComponents.div`
```
Comment (jamesjpedi) [inline src/pages/Flight.js:8]: Move styles to separate file

File: src/pages/Activities.js [hunk 9]
```diff
@@ -0,0 +1,24 @@
+import React from 'react'
+import styledComponents from "styled-components";
+import SearchButton from "../components/SearchButton/SearchButton";
+
+const MainDiv = styledComponents.div`
```
Comment (jamesjpedi) [inline src/pages/Activities.js:5]: Move styles to separate file

File: src/pages/Flight.js [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+  border-radius: 8px;
+  box-shadow: 0 1px 5px 0 rgb(0 0 0 / 10%);
+  background-color: #fff;
+  padding: 60px 20px 0;
+`;
+
+const Flight = (props) => {
+  let desc = props.labelFor === 'flight'?'Book International and Domestic Flights':'Book Charter Planes in India';
+  return (
+    <MainDiv>
```
Comment (jamesjpedi) [inline src/pages/Flight.js:21]: Main div is shared between different components. Check possibility of move to single component with custom / additional styles

File: src/components/SearchBox/SearchBox.js [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+  background-repeat: no-repeat;
+  display: inline-block;
+  background-size: 200px 650px;
+  font-size: 0px;
+  flex-shrink: 0;
+`
+const SearchBox = () => {
+  return (
+    <SearchBoxDiv>
+      <LocationWidget
```
Comment (jamesjpedi) [inline src/components/SearchBox/SearchBox.js:52]: Indentation is off

File: src/components/SearchBox/SearchBox.js [hunk 12]
```diff
@@ -0,0 +1,81 @@
+import React from 'react'
```
Comment (jamesjpedi) [inline src/components/SearchBox/SearchBox.js:1]: Better use file name as index.js as its single file inside folder name same a the component.

File: src/components/SearchButton/SearchButton.js [hunk 13]
```diff
@@ -0,0 +1,39 @@
+import React from 'react'
```
Comment (jamesjpedi) [inline src/components/SearchButton/SearchButton.js:1]: Formating off
Moved styles to global

File: src/components/SearchFare/SearchFare.js [hunk 14]
```diff
@@ -0,0 +1,50 @@
+import React from "react";
+import FareWidget from "../../widgets/searchFare/FareWidget";
+import TrendingWidget from "../../widgets/searchFare/TrendingWidget";
+import styledComponents from "styled-components";
+const FareDiv = styledComponents.div`
+  display: flex;
+  // justify-content: space-around;
```
Comment (jamesjpedi) [inline src/components/SearchFare/SearchFare.js:7]: Do not include commented code

File: src/components/SearchFare/SearchFare.js [hunk 15]
```diff
@@ -0,0 +1,50 @@
+import React from "react";
+import FareWidget from "../../widgets/searchFare/FareWidget";
+import TrendingWidget from "../../widgets/searchFare/TrendingWidget";
+import styledComponents from "styled-components";
+const FareDiv = styledComponents.div`
+  display: flex;
+  // justify-content: space-around;
+`
+const data=[
```
Comment (jamesjpedi) [inline src/components/SearchFare/SearchFare.js:9]: Move constants to global file

File: src/components/SearchFare/SearchFare.js [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+        data={data}
+      />
+      <TrendingWidget 
+      locations={[
+        'Pune --> Delhi',
+        'Bangalore --> Chennai'
+      ]}
+        />
+    </FareDiv>
+
```
Comment (jamesjpedi) [inline src/components/SearchFare/SearchFare.js:46]: Formatting off

File: src/components/SearchTrip/SearchTrip.js [hunk 17]
```diff
@@ -0,0 +1,35 @@
+import React from "react";
+import styledComponents from "styled-components";
+import TripWidget from "../../widgets/searchTrip/TripWidget";
+const TripDiv = styledComponents.div`
+  display: flex;
+`
+const tripLabel = [
```
Comment (jamesjpedi) [inline src/components/SearchTrip/SearchTrip.js:7]: Move to global file

**Review by jamesjpedi — COMMENTED** [review]
Try to do the mentioned comments through the entire codebase. Not just only the mentioned place / file.

File: src/index.js (hunk 1 above)
Comment (arunkjojo) [reply src/index.js:9]: Ok.

File: src/pages/Flight.js (hunk 10 above)
Comment (arunkjojo) [reply src/pages/Flight.js:21]: Changed.

**Review by arunkjojo — COMMENTED** [review]
current review comments all are done.

**Pull request merged**